### PR TITLE
feat(browser): add support for the new headless mode in BrowserConfig

### LINKED ISSUE #183: Use new headless mode.
Currently library passes ``--headless`` switch instead of ``--headless=new``
https://github.com/mattsse/chromiumoxide/blob/88470cabec15ffc2a91dc5579ec142e921d59eab/src/browser.rs#L823
Old headless, according to the chrome's blog, is separate browser implementation which leads to being detected by antibots like cloudflare waf.
https://developer.chrome.com/articles/new-headless/

I think we could just explicitly use ``--headless=new``.

### PR DESCRIPTION
**Pull request by starrify**
resolves #183 (kind of)

### CODE REVIEW

File: src/browser.rs [hunk 1]
```diff
@@ -559,11 +559,18 @@ async fn ws_url_from_output(
     }
 }
 
+#[derive(Debug, Clone)]
+enum HeadlessMode {
+    False,
+    True,
+    New,
+}
```
Comment (mattsse) [inline src/browser.rs:567]: can we document these?

**Review by mattsse — CHANGES_REQUESTED** [review]

File: src/browser.rs (hunk 1 above)
Comment (starrify) [reply src/browser.rs]: Hi @mattsse sure and it's now added. Sorry for not having included any description from the start.

File: src/browser.rs [hunk 2]
```diff
@@ -559,11 +559,21 @@ async fn ws_url_from_output(
     }
 }
 
+#[derive(Debug, Clone)]
+enum HeadlessMode {
+    /// The "headful" mode.
+    False,
+    /// The old headless mode (default).
+    True,
```
Comment (mattsse) [inline src/browser.rs:567]: this should be #[default]

File: src/browser.rs [hunk 3]
```diff
@@ -686,7 +696,12 @@ impl BrowserConfigBuilder {
...
     pub fn with_head(mut self) -> Self {
-        self.headless = false;
+        self.headless = HeadlessMode::False;
+        self
+    }
+
+    pub fn new_headless_mode(mut self) -> Self {
+        self.headless = HeadlessMode::New;
         self
     }
```
Comment (mattsse) [inline src/browser.rs:706]: we want a new function that provides the `HeadlessMode` as argument

File: src/browser.rs [hunk 4]
```diff
@@ -559,11 +559,21 @@ async fn ws_url_from_output(
...
 }
 
+#[derive(Debug, Clone)]
```
Comment (mattsse) [inline src/browser.rs:562]:
```suggestion
#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
```

**Review by mattsse — CHANGES_REQUESTED** [review]
last nits

**Comment by starrify** [comment]
Hi @mattsse , thanks a lot for the recommendations! I have the request changes applied accordingly.

**Review by mattsse — APPROVED** [review]

**Pull request merged**